cai_causal_graph/time_series_causal_graph.py

```diff
@@ -47,13 +47,40 @@ def wrapper(self: TimeSeriesCausalGraph, *args, **kwargs) -> Any:
         self._minimal_graph = None
         self._summary_graph = None
         self._stationary_graph = None
-        self._maxlag = None
         self._variables = None
         return function
 
     return wrapper
 
 
+def get_min_lag(ts_graph: TimeSeriesCausalGraph) -> Optional[int]:
+    """
+    Get the minimum lag from a TimeSeriesCausalGraph.
+
+    :param ts_graph: The time series causal graph.
+    :return: The minimum lag.
+    """
+    assert isinstance(
+        ts_graph, TimeSeriesCausalGraph
+    ), f'Expected ts_graph to be a TimeSeriesGraph, got type {type(ts_graph)}.'
+    nodes = ts_graph.get_nodes()
+    return min([node.time_lag for node in nodes]) if len(nodes) > 0 else None
```

Comment:
Minor point but this builds entire list first. `min(node.time_lag for node in nodes)` doesn't require constructing the full list.